### ISSUE TITLE
Not able to open Compound Templates for DDI modelling

### ISSUE THREAD

**Issue by arketkar** [Open-Systems-Pharmacology/PK-Sim#1477]
I am trying to open the default compound templates for performing DDI modelling. However getting the error message as follows.

Application:
PK-Sim® 8.0.22

There is no row at position 0.

Stack trace:
```
at System.Data.RBTree`1.GetNodeByIndex(Int32 userIndex)
   at System.Data.DataRowCollection.get_Item(Int32 index)
   at PKSim.Infrastructure.ORM.DAS.DASDataRowCollection.ItemByIndex(Int32 Index)
   at PKSim.Infrastructure.Extensions.DASExtensions.ExecuteQueryForSingleRow(DAS connection, String query)
   at PKSim.Infrastructure.Services.TemplateTaskQuery.loadTemplateBy(TemplateDatabaseType templateDatabaseType, String templateName, String templateType, DAS connection)
   at PKSim.Infrastructure.Services.TemplateTaskQuery.addReferencesToTemplate(Template template, DAS connection, List`1 loadedReferenceNames)
   at PKSim.Infrastructure.Services.TemplateTaskQuery.AllTemplatesFor(TemplateDatabaseType templateDatabaseType, TemplateType templateType)
   at PKSim.Infrastructure.Services.TemplateTaskQuery.AllTemplatesFor(TemplateType templateType)
   at PKSim.Presentation.Presenters.TemplatePresenter.LoadFromTemplate[T](TemplateType templateType)
   at PKSim.Presentation.Services.BuildingBlockTask.LoadFromTemplate[TBuildingBlock](PKSimBuildingBlockType buildingBlockType)
   at PKSim.Presentation.Services.BuildingBlockTask`1.LoadFromTemplate(PKSimBuildingBlockType buildingBlockType)
   at PKSim.Presentation.Services.BuildingBlockTask`1.LoadFromTemplate()
   at PKSim.Presentation.Services.BuildingBlockTask`1.LoadSingleFromTemplate()
   at PKSim.Presentation.UICommands.LoadBuildingBlockFromTemplateUICommand`2.Execute()
   at OSPSuite.Presentation.UICommands.UICommandExtensions.ExecuteWithinExceptionHandler(IUICommand uiCommand, IEventPublisher eventPublisher, IChangePropagator changePropagator)
```

**Comment by Yuri05** [comment]
Please describe in more detail what you are doing.
As far as I understand you are trying to load one or more compounds from Predefined compound templates?
Which compound(s)?

**Comment by arketkar** [comment]
Hi Juri,

Thanks for your reply. I am trying to load Midazolam compound template. The moment I click on Load from Template, I receive an error.

I am also facing the same problem while I am trying to save my compound by clicking ‘Save as template’.

This used to work earlier though. I do not know what has happened since last few days with this option. I am facing this type of error in a new project and also in an earlier saved project.

Thanks and regards.
Anant

**Comment by Yuri05** [comment]
Can you try the following:
* close PK-Sim
* go to **C:\ProgramData\Open Systems Pharmacology\PK-Sim\8.0**
* temporarily rename the file **PKSimTemplateDBSystem.templateDbSystem**
* download https://github.com/Open-Systems-Pharmacology/PK-Sim/raw/4f17a9304e63f749f8b19e3f7f05dcda1664d89d/src/Db/TemplateDB/PKSimTemplateDBSystem.templateDbSystem and save it under  **C:\ProgramData\Open Systems Pharmacology\PK-Sim\8.0**
* start PK-Sim and try to load Midazolam or any other predefine compound from template

Did it work?

**Comment by arketkar** [comment]
Hi Juri,

There was no such file in the path you mentioned. Then I searched through the ‘Options’ in PK-Sim 8, where I found the following path under ‘Template database path’.

C:\Users\anant.ketkar\AppData\Roaming\Open Systems Pharmacology\PK-Sim\8.0\PKSimTemplateDBUser.templateDbUser

I renamed that file and copied the file shared by you. It is working fine now. I guess the file which was there got corrupt or so…

Thank you so much for your quick support. Appreciate that.

Regards.
Anant
